albatross/analysis.py

```diff
@@ -0,0 +1,56 @@
+"""
+Provides analysis tools for wind data.
+"""
+
+import matplotlib.pyplot as plt
+from pandas import DataFrame
+
+
+def boxplot(data, fields=None, labels=None, **box_kwargs):
+    """
+    Draws boxplots of wind speeds.
+
+    Args:
+      data (DataFrame): wind data
+      fields (:obj:`list` of :obj:`str`, optional): a list of columns to include from the
+        given `data`. If none are provided, these will be inferred using any columns in
+        `data` with the prefix `'windspeed_'`.
+      labels (:obj:`list` of :obj:`str`, optional): a list of labels to use. If none are
+        provided, they will use the same names as `fields`. If no `fields` or `labels`
+        are provided, they will both be inferred using the same strategy as `fields`, but
+        taking the suffix after `'windspeed_'`. e.g. `'windspeed_90m'` -> `'90m'`
+      box_kwargs (dict, optional): additional parameters for `matplotlib.pyplot.boxplot`
+
+    Returns:
+      tuple: A tuple (fig, ax) consisting of a `matplotlib.figure.Figure` and
+      `matplotlib.axes.Axes`.
+    """
+    assert isinstance(data, DataFrame), '"data" must be a DataFrame'
+    if fields:
+        assert isinstance(fields, list), '"fields" must be a list or None'
+        msg = '"fields" elements must be strings'
+        assert all([isinstance(f, str) for f in fields]), msg
+    if labels:
+        assert isinstance(labels, list), '"labels" must be a list or None'
+        msg = '"labels" elements must be strings'
+        assert all([isinstance(label, str) for label in labels]), msg
+
```

Comment:
With a quick google search I found that `if var:` is supposedly equal to `if bool(var)`, but the variables in your case are either a list of strings or `None`. Could you explain how these if statements are supposed to function? I haven't used them this way before

Reply:
Basically, in Python values like `None`, `0`, and `[]` are falsy (you can test some out yourself by doing `bool([])`, etc.), so you can implicitly use them in conditionals. Personally, I only use this shortcut when it's known that the value will either be `None` or something truthy, and will opt to be more explicit otherwise (`if len(my_list) == 0`, `if val == 0`, etc.)